### PR TITLE
Cluster Profile Tutorial

### PR DESCRIPTION
**Pull request by Alez87**
Added the tutorial draft "Deploy a Kubernetes Cluster with Palette."

🔍 [Preview](https://palette-app-tutorial--docs-spectrocloud.netlify.app/knowledgebase/tutorials/dev-engine/deploy-k8s-cluster)

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1151#" title="Author self-approved">Alez87</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
Hi @Alez87. Thanks for your PR.

I'm waiting for a [spectrocloud](https://github.com/orgs/spectrocloud/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/spectrocloud/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Review by karl-cardenas-coding — COMMENTED** [review]
For some reason, I lost access to all my pending comments, so I have to display them. THe review is still ongoing.

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md:1586]: This can all be reduced to a single command `terraform apply --auto-approve`

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md:1593]: This step is not needed with Terraform. If you keep everything in the same folder, the cluster will receive the new update. This problem will go away when you simplify everything under two main tabs UI workflow and Terraform.

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md:1663]:
```suggestion
Use the destroy command to remove all the resources you created through Terraform.
```

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md:1677]:
```suggestion
terraform destroy --auto-approve
```

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (karl-cardenas-coding) [inline content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md:1680]:
```suggestion
The deletion process will take a few minutes. 
```
Let's also include the output that shows the count of resources removed and let's add a note about force delete. 

CleanShot 2023-04-07 at 10 09 48

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
Great start to the tutorial. There is a lot of feedback provided but that is expected with this being your first tutorial. I'm gonna reach out to you in the Slack channel and explain a lot of the feedback.

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
Great start to the tutorial. There is a lot of feedback provided but that is expected with this being your first tutorial. I'm gonna reach out to you in the Slack channel and explain a lot of the feedback.

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md:870]: For AWS I'm forced to leave the K8s version to "1.21.14" otherwise is not compatible with Amazon Cloud Native Storage (csi-aws)

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md:1061]: Done

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md]: Done

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md:1158]: Done

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md:1]: Done

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md]: Done

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md:1420]: Done

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md:1186]: All the Terraform files moved to one folder and modified the config files accordingly.

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md:1707]: Removed for Terraform

**Review by Alez87 — COMMENTED** [review]
review finished

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md]: done

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md]: added an env variable

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md]: done

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md]: done

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md]: done

File: content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/10-dev-engine/03-deploy-k8s-cluster.md]: done

File: content/docs/18-knowledgebase/01-tutorials/3-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/3-deploy-k8s-cluster.md]: Modified

File: content/docs/18-knowledgebase/01-tutorials/3-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/3-deploy-k8s-cluster.md]: Modified

File: content/docs/18-knowledgebase/01-tutorials/3-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/3-deploy-k8s-cluster.md]: Modified

File: content/docs/18-knowledgebase/01-tutorials/3-deploy-k8s-cluster.md
Comment (Alez87) [reply content/docs/18-knowledgebase/01-tutorials/3-deploy-k8s-cluster.md]: Modified

**Comment by karl-cardenas-coding** [comment]
Tutorial moved over to #1267

**Pull request closed without merging**
